### ISSUE TITLE
RTD error

### ISSUE THREAD

**Issue by adam-a-a** [watertap-org/watertap-reflo#82]
I tried troubleshooting this failed readthedocs check but am at a loss: 

```
Running Sphinx v7.2.6

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/watertap-seto/envs/81/lib/python3.10/site-packages/sphinx/registry.py", line 447, in load_extension
    mod = import_module(extname)
  File "/home/docs/.asdf/installs/python/3.10.13/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'sphinx_rtd_theme'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/watertap-seto/envs/81/lib/python3.10/site-packages/sphinx/cmd/build.py", line 293, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/home/docs/checkouts/readthedocs.org/user_builds/watertap-seto/envs/81/lib/python3.10/site-packages/sphinx/application.py", line 233, in __init__
    self.setup_extension(extension)
  File "/home/docs/checkouts/readthedocs.org/user_builds/watertap-seto/envs/81/lib/python3.10/site-packages/sphinx/application.py", line 406, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/watertap-seto/envs/81/lib/python3.10/site-packages/sphinx/registry.py", line 450, in load_extension
    raise ExtensionError(__('Could not import extension %s') % extname,
sphinx.errors.ExtensionError: Could not import extension sphinx_rtd_theme (exception: No module named 'sphinx_rtd_theme')

Extension error:
Could not import extension sphinx_rtd_theme (exception: No module named 'sphinx_rtd_theme')
```

It is not clear why `sphinx_rtd_theme` suddently isn't recognized as an import anymore. I thought it might have something to do with some of our renaming of "seto" to "reflo" but so far that doesn't seem to be the issue (unless I missed something). See PR #81 where I have been probing this.

**Comment by adam-a-a** [comment]
@lbianchi-lbl FYI

**Comment by lbianchi-lbl** [comment]
@adam-a-a I've looked into the most recent passing and failing builds for #81, and this is the diff of the build logs: https://www.diffchecker.com/UV18GwhN/ (left-hand side: failing, right-hand side: passing).

- It looks like [ddf66a2](https://github.com/watertap-org/watertap-seto/pull/81/commits/ddf66a2034138db3ac64c5b2d441b1e57f44f5bc), i.e. adding `sphinx-readthedocs-theme` to `requirements-dev.txt`, did the trick, i.e. the package gets installed during the RTD build
- I see that `sphinx-readthedocs-theme` was in the `[dev]` extras_require target in `setup.py`, and that you were trying to e.g. pin the version to see if it had any effect, but I was confused as to why it didn't seem to make any difference
- After checking `setup.py` and `requirements-dev.txt`, I believe the reason is that `requirements-dev.txt` doesn't specify the `[dev]` in the `pip install -e .` line; therefore, all those dependencies are ignored
- From `git blame`, it looks like the `[dev]` target was added in https://github.com/watertap-org/watertap-seto/commit/929a11e4bf308aefae2e698c7c9b26d35dd41aed, and we (starting with myself as the reviewer) didn't realize those dependencies weren't being picked up by the default installation command which relies on `requirements-dev.txt`
- The remedy should be to just change `pip install -e .` to `pip install -e .[dev]` in `requirements-dev.txt`
- This doesn't explain why the RTD build suddenly stopped working (i.e. why it was working before, if the `[dev]` dependencies were being ignored); my guess is that maybe `sphinx-readthedocs-theme` was available by default on the RTD build servers until recently

**Comment by adam-a-a** [comment]
Thanks for the assessment, @lbianchi-lbl . I noticed that my last commit did the trick as you pointed out, but I was going to ask you for a better fix, which you've provided. Thanks again!

**Comment by adam-a-a** [comment]
I incorporated fixes as need within PRs #76  and #77

**Comment by adam-a-a** [comment]
I incorporated fixes as need within PRs #76  and #77
